### PR TITLE
Feature/rushfiles

### PR DESCRIPTION
**Pull request by Makinami**
My name is Wiktor Klonowski and I am sending this pull request on behalf of Tsukaeru.net.

## Purpose
As discussed via email and video conference in October last year, we would like to add to GakuNin RDM a support for our file storage solution: RushFIles (FileBako in Japan). This PR (and corresponding PR in waterbutler repository: https://github.com/RCOSDP/RDM-waterbutler/pull/36) implements an addon for this integration.

At the moment, addon integrates only with user panel. Integration with organization's administrator control panel might come in the future, but is out of scope of this PR.

## Changes
Add support for a new file storage service called RushFiles (FileBako in Japan).

## QA Notes

  - Does this change require a data migration? If so, what data will we migrate?
    Yes, but only the basic addition to the database to support new addon. Although we've had some problems with incompatibility with broken migration chain with wiki addon.
  - What is the level of risk?
    Should be minimal.
    - Any permissions code touched?
      No.
    - Is this an additive or subtractive change, other?
      Purely additive. It introduces a new addon and it's static files/translations, without modifying any of the base source code.
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?)
    This PR adds a new addon usable by user. The addon does not provide functionality in admin control panel at the moment, so testing should be done through user's UI.
    - If verifying through API, what's the new version? Please include the endpoints in PR notes or Dev docs.
      No direct changes to the API.
  - What features or workflows might this change impact?
    Added support for a new file storage.
  - How will this impact performance?
    Impact should be minimal, as we are only loading one more addon.

Please let us know if you need a RushFiles/FileBako account for testing.

## Documentation
Depending on whether such list is provided on RCOSDP page, a list of supported file storage solutions might need updating.

The rudimentary documentation for RushFiles API can be found here: https://wiki.rushfiles.com/api . For any questions, you may contact support@tsukaeru.net .

## Side Effects
No known side effects.

## Ticket
N/A

## Notes

1. This feature branch was initially created in October from the develop, which was still using Python 2.7 back then, and developed as such. Before sending this PR, I have rebased the branch to the newest version of develop branch. Unfortunately, after the rebase I can't start GakuNin RDM system. Problems are with incompatible base requirements and/or missing linux libraries in docker image. 
Despite our best effort, we couldn't solve this problem without internal knowledge of the entire system or intrusive changes. Because of this, the addon wasn't tested on the latest version of branch. However, as it does not use deprecated Python 2.7 features and works similarly to other storage addons, I don't believe there will be any problem with running it on Python 3.6. Should there be bugs or if you know how to build and run newest develop branch, please let us know.
2. The latest commit (91117ceeec2c4c319798dfc062414598e8f2bc29), changes user facing branding of the addon from RushFiles to FileBako. This is because we are the partner and reseller of RushFiles in Japan and in Japan it is known under the FileBako name: https://www.tsukaeru.net/file-bako / https://cloudfile.jp/

### CODE REVIEW

File: addons.json [hunk 1]
```diff
@@ -92,7 +94,8 @@
         "nextcloudinstitutions",
         "iqbrims",
         "dropboxbusiness",
-        "onedrivebusiness"
+        "onedrivebusiness",
+        "partial"
```
Comment (hide24) [inline addons.json]: `addons_commentable`  is array of AddonAppConfig.short_name.
It should be `rushfiles`. (if comentable)

File: framework/addons/data/addons.json [hunk 2]
```diff
@@ -679,10 +679,40 @@
                 "status": "none",
                 "text": "Metadata information will not be registered."
             }
+        },
+        "RushFiles": {
```
Comment (hide24) [inline framework/addons/data/addons.json]: https://github.com/tsukaeru/RDM-osf.io/blob/feature/rushfiles/framework/addons/data/addons.json#L683
https://github.com/tsukaeru/RDM-osf.io/blob/feature/rushfiles/addons/rushfiles/apps.py#L12
https://github.com/tsukaeru/RDM-osf.io/blob/2197b054c0ec41fac26f89b69019b3c670ccb53f/addons/rushfiles/static/node-cfg.js#L6

Those symbols must match. (`FileBako` or `RushFiles`)

**Review by hide24 — CHANGES_REQUESTED** [review]

File: addons.json (hunk 1 above)
Comment (Makinami) [reply addons.json]: Fixed to `rushfiles`. Sorry for missing it.

File: framework/addons/data/addons.json (hunk 2 above)
Comment (Makinami) [reply framework/addons/data/addons.json]: Changed here to `FileBako`.

**Review by hide24 — APPROVED** [review]

**Comment by hide24** [comment]
merged!

**Pull request closed without merging**